bentoml/_internal/io_descriptors/json.py

```diff
@@ -269,11 +285,63 @@ async def to_http_response(
         if ctx is not None:
             res = Response(
                 json_str,
-                media_type=MIME_TYPE_JSON,
+                media_type=self._mime_type,
                 headers=ctx.response.metadata,  # type: ignore (bad starlette types)
                 status_code=ctx.response.status_code,
             )
             set_cookies(res, ctx.response.cookies)
             return res
         else:
-            return Response(json_str, media_type=MIME_TYPE_JSON)
+            return Response(json_str, media_type=self._mime_type)
+
+    async def from_proto(self, request: pb.Request) -> JSONType:
+        from google.protobuf.json_format import MessageToDict
+
+        if request.HasField("json"):
+            json_obj = MessageToDict(request.json, preserving_proto_field_name=True)
+
+            if self._pydantic_model:
+                try:
+                    return self._pydantic_model.parse_obj(json_obj)
+                except pydantic.ValidationError as e:
+                    raise UnprocessableEntity(f"Invalid JSON input received: {e}")
+        elif request.HasField("raw_bytes_contents"):
+            content = request.raw_bytes_contents
+            if self._pydantic_model:
+                try:
+                    return self._pydantic_model.parse_raw(content)
+                except pydantic.ValidationError as e:
+                    raise UnprocessableEntity(f"Invalid JSON input received: {e}")
+
+            try:
+                json_obj = json.loads(content)
+            except json.JSONDecodeError as e:
+                raise UnprocessableEntity(f"Invalid JSON input received: {e}")
```

Comment:
We shouldn't be using unprocessable entity here; it should probably be a 400 bad request.